CSharp/BatchExplorer/ViewModels/CreateCertificateViewModel.cs

```diff
@@ -0,0 +1,165 @@
+            }
+            set
+            {
+                this.filePath = value;
+                this.FirePropertyChangedEvent("FilePath");
+                this.FirePropertyChangedEvent("IsCreateCertificateButtonEnabled");
+            }
+        }
+
+        private string password;
+
+        public string Password
+        {
+            get
+            {
+                return this.password;
+            }
+            set
+            {
+                this.password = value;
+                this.FirePropertyChangedEvent("Password");
+                this.FirePropertyChangedEvent("IsCreateCertificateButtonEnabled");
+            }
+        }
+
+        public bool IsCreateCertificateButtonEnabled
+        {
+            get
+            {
+                if (!FilePathExists())
+                {
+                    return false;
+                }
+
+                var format = Format();
+
+                if (format == null)
+                {
+                    return false;
+                }
+
+                if (format.Value == CertificateFormat.Pfx && string.IsNullOrEmpty(Password))
+                {
+                    return false;
+                }
+
+                return true;
+            }
+        }
+
+        public CommandBase CreateCertificate
+        {
+            get
+            {
+                return new CommandBase(
+                    async (o) =>
+                    {
+                        try
+                        {
+                            this.IsBusy = true;
+                            await this.CreateCertificateAsync();
+                        }
+                        finally
+                        {
+                            this.IsBusy = false;
+                        }
+                    }
+                );
+            }
+        }
+
+        private bool FilePathExists()
+        {
+            if (String.IsNullOrWhiteSpace(FilePath))
+            {
+                return false;
+            }
+
+            try
+            {
+                return System.IO.File.Exists(FilePath);
+            }
+            catch
+            {
+                return false;
+            }
+        }
+
+        private CertificateFormat? Format()
+        {
+            var fileExtension = System.IO.Path.GetExtension(FilePath);
+            var comparer = StringComparer.OrdinalIgnoreCase;
+
+            if (comparer.Equals(".pfx", fileExtension))
+            {
+                return CertificateFormat.Pfx;
+            }
+            if (comparer.Equals(".cer", fileExtension))
+            {
+                return CertificateFormat.Cer;
+            }
+            return null;
+        }
+
+        private async Task CreateCertificateAsync()
+        {
+            try
+            {
+                var certificateFormat = Format();
+
+                if (certificateFormat == null)
+                {
+                    return;  // shouldn't have got here
```

Comment:
As this is a "can't happen" it's not clear what exception to throw - the user can't do anything about it.  I've added a Debug.Assert instead.